### DISCUSSION TITLE
How can I hide fields(like connection string) from the logs and also from the elsa studio dashboard?

### DISCUSSION THREAD

**Discussion by jordanpedro** [elsa-workflows/elsa-core#4995] [Q&A]
We have some sensitive information in our workflows that we do not wants to let it stored anywhere. Such as connection string and other sensitive data. Today we get these information from the secrets created in our kubernetes cluster (through a custom javascript function created to take environment variables using C#). I just cant let sensitive data being stored. i would like to point out thats it is an extremily useful and important feature. If anyone knows how i can achieve that in elsa v3, please lemme know

**Comment by sfmskywalker** [comment]
Elsa 3 currently lacks this capability, but is something we also need for another customer relatively soon (Q2).

**Comment by sfmskywalker** [comment]
I was looking for the issue where we are keeping track of this feature, but I couldn't find it so I created a new one: https://github.com/elsa-workflows/elsa-core/issues/4994

**Comment by glime-ai** [comment]
To handle sensitive information such as connection strings and other secrets in Elsa v3 without storing them in the workflow definitions, you can use environment variables or secrets management systems that are accessible during the workflow execution. Since you are already using Kubernetes, you can continue to leverage Kubernetes secrets.

In Elsa v3, you can access environment variables directly within your workflow expressions using C# script or JavaScript. You can create custom activities that fetch secrets at runtime from your Kubernetes cluster or any other secrets management system you use, ensuring that sensitive data is not stored within the workflow definitions.

Here's a general approach you can take:

1. Create a custom activity that retrieves the sensitive data at runtime from the secrets store (Kubernetes secrets, Azure Key Vault, AWS Secrets Manager, etc.).
2. Use this custom activity within your workflows wherever you need to use the sensitive data.
3. Ensure that the custom activity does not persist the sensitive data in the workflow logs or execution context.

For accessing environment variables in Elsa v3, you can use the `env` function in JavaScript expressions or access the `Environment` class in C# expressions.

Please note that while I can provide this general guidance, the specific implementation details will depend on your application's architecture and the secrets management solution you are using.

If you need further assistance with implementing this feature or if you have specific questions about custom activities in Elsa v3, I recommend reaching out to the Elsa community or seeking professional consultation as mentioned in the GitHub issue you referenced.
